src/Avalonia.Visuals/Media/HsvColor.cs

```diff
@@ -0,0 +1,514 @@
+﻿// Color conversion portions of this source file are adapted from the WinUI project. 
+// (https://github.com/microsoft/microsoft-ui-xaml) 
+// 
+// Licensed to The Avalonia Project under MIT License, courtesy of The .NET Foundation.
+
+using System;
+using Avalonia.Utilities;
+
+namespace Avalonia.Media
+{
+    /// <summary>
+    /// Defines a color using the hue/saturation/value (HSV) model.
+    /// </summary>
+#if !BUILDTASK
+    public
+#endif
+    readonly struct HsvColor : IEquatable<HsvColor>
+    {
+        /// <summary>
+        /// Represents a null <see cref="HsvColor"/> with zero set for all channels.
+        /// </summary>
+        public static readonly HsvColor Empty = new HsvColor();
+
+        ////////////////////////////////////////////////////////////////////////
+        // Constructors
+        ////////////////////////////////////////////////////////////////////////
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HsvColor"/> struct.
+        /// </summary>
+        /// <param name="hue">The Hue channel value in the range from 0..360.</param>
+        /// <param name="saturation">The Saturation channel value in the range from 0..1.</param>
+        /// <param name="value">The Value channel value in the range from 0..1.</param>
+        public HsvColor(
+            double hue,
+            double saturation,
+            double value)
+        {
+            A = 1.0;
+            H = MathUtilities.Clamp((double.IsNaN(hue) ? 0 : hue), 0, 360);
+            S = MathUtilities.Clamp((double.IsNaN(saturation) ? 0 : saturation), 0, 1);
+            V = MathUtilities.Clamp((double.IsNaN(value) ? 0 : value), 0, 1);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HsvColor"/> struct.
+        /// </summary>
+        /// <param name="hue">The Hue channel value in the range from 0..360.</param>
+        /// <param name="saturation">The Saturation channel value in the range from 0..1.</param>
+        /// <param name="value">The Value channel value in the range from 0..1.</param>
+        /// <param name="alpha">The Alpha (transparency) channel value in the range from 0..1.</param>
+        public HsvColor(
+            double hue,
+            double saturation,
+            double value,
+            double alpha)
```

Comment:
The last thing that was caught in an internal review was the order of parameters here. In the Color constructor and FromAhsv, FromArgb, etc, Alpha channel is the first parameter. This is somewhat standard in .NET and follows with pre-multiplied alpha ordering as well.

The reason alpha was put at the end is due to the way Windows Community Toolkit methods work. This type was originally a drop-in replacement for some of those methods: https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/b39d70f6ea4d739a470768c103c8af2c5b1e02ef/Microsoft.Toolkit.Uwp/Helpers/ColorHelper.cs#L312

This is the same as encountered here in the `HsvColor.ToRgb()` and `HsvColor.FromRgb()` methods because alpha channel is optional and therefore must be at the end.

As this would be a pretty bad breaking change in the future (undetectable errors), we need to be clear on parameter ordering here. @jmacato @maxkatz6 What are your thoughts? Should alpha be first always except for methods where it is optional?

Reply:
Another reason to keep alpha at the end is because there are two constructors: one with alpha, one without. With alpha at the end it doesn't change the position of the HSV channels and therefore there would theoretically be fewer errors switching between constructors.

Reply:
@maxkatz6 @MarchingCube @jmacato Any preference with this?

Reply:
use alpha, no need to follow UWP/WPF conventions if it doesnt make any sense in our "local" context.

Reply:
follow the conventions of Avalonia's Color class since Hsv is just another form of it.

Reply:
Alright will put it at the front as flagged to match Color. This will standardize the `HsvColor(A,H,S,V)` and `Color(A,R,G,B)` constructors as well as follow the ordering in `Color.FromArgb()` and `Hsv.FromAhsv()`. 

Just realize this is now different than the ordering in the new To/FromRGB methods within HsvColor because the alpha is optional and must be last due to c# rules.

Reply:
Unless there are objections I'm planning to remove both the `Color(H,S,V)` constructor and the `HsvColor.FromHsv(H,S,V)` static method.

 1. As stated, there is concern that parameters ordering differences could cause issues if developers aren't careful. AHSV vs. HSV is easy to be off-by-one in channel ordering when moving from HSV to AHSV methods for example.
 2. There are no situations where AHSV parameter equivalents can't be used in place of HSV versions. Worse-case, the caller will need to provide 1.0 for the alpha value.
 3. Color doesn't have RBG-only constructors/methods. Everything takes an alpha channel value as well.

Reply:
I've removed the code as mentioned. I wish the core team was a little more vocal about API-related discussions like this. These things are important to discuss but I realize a bit boring/trivial.

Marking as resolved